<a href="https://colab.research.google.com/github/ahmedaouadi91/project-2/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd
import numpy as np

# Dataset 1 : Adult income

In [ ]:
df1 = pd.read_csv('/content/drive/MyDrive/adult.csv')
df1.info()
df1.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   age              48842 non-null  int64 
 1   workclass        48842 non-null  object
 2   fnlwgt           48842 non-null  int64 
 3   education        48842 non-null  object
 4   educational-num  48842 non-null  int64 
 5   marital-status   48842 non-null  object
 6   occupation       48842 non-null  object
 7   relationship     48842 non-null  object
 8   race             48842 non-null  object
 9   gender           48842 non-null  object
 10  capital-gain     48842 non-null  int64 
 11  capital-loss     48842 non-null  int64 
 12  hours-per-week   48842 non-null  int64 
 13  native-country   48842 non-null  object
 14  income           48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

1. Source of data

* https://www.kaggle.com/datasets/wenruliu/adult-income-dataset

2. Brief description of data

* An individual’s annual income results from various factors. Intuitively, it is influenced by the individual’s education level, age, gender, occupation, and etc.

3. What is the target?

* The target is the income column

4. What does one row represent? (A person? A business? An event? A product?)

* one row represent A person (an adult)

5. Is this a classification or regression problem?

* it is a classification problem

6. How many features does the data have?

* Data has 15 Features
7. How many rows are in the dataset?

* Data has 48842 rows

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
* Data is  little big ,we can meet so many duplicated rows ,and may be the data will be imbalanced ,and that can soak the model  

# Dataset 2 : Crop Recommendation

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/Crop_recommendation.csv')
df2.info()
df2.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2200 entries, 0 to 2199
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   N            2200 non-null   int64  
 1   P            2200 non-null   int64  
 2   K            2200 non-null   int64  
 3   temperature  2200 non-null   float64
 4   humidity     2200 non-null   float64
 5   ph           2200 non-null   float64
 6   rainfall     2200 non-null   float64
 7   label        2200 non-null   object 
dtypes: float64(4), int64(3), object(1)
memory usage: 137.6+ KB


N   P   K  temperature   humidity        ph    rainfall label
0  90  42  43    20.879744  82.002744  6.502985  202.935536  rice
1  85  58  41    21.770462  80.319644  7.038096  226.655537  rice
2  60  55  44    23.004459  82.320763  7.840207  263.964248  rice
3  74  35  40    26.491096  80.158363  6.980401  242.864034  rice
4  78  42  42    20.130175  81.604873  7.628473  262.717340  rice

1. Source of data

* https://www.kaggle.com/datasets/atharvaingle/crop-recommendation-dataset

2. Brief description of data

* A crop yield results from various factors. Intuitively, it is influenced by the ratio of Nitrogen ,Phosphorous and Potassium content in soil ,temperature ,humidity ,ph value of the soil and rainfall.
this dataset allows the users to build a predictive model to recommend the most suitable crops to grow in a particular farm based on various parameters.
3. What is the target?
* The target is the label columns

4. What does one row represent? (A person? A business? An event? A product?)
* one row represent a farm
5. Is this a classification or regression problem?
* this is a classification model

6. How many features does the data have?
* Data has 8 feature

7. How many rows are in the dataset?
* Data has 2200 rows

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
*  Data is  little small , may be the data will be imbalanced,duplicated rows ,and the risk of underfitting model
